### ISSUE TITLE
bcftools view exclude sample trim-alt-alleles  | bcf_remove_allele_set: Assertion `nori==nA_ori' failed.

### ISSUE THREAD

**Issue by NeillGibson** [samtools/bcftools#404]
Hi,

When I try to exclude a sample from a vcf file and to trim the alt alleles  I run in to an error. 

The command I am trying is 

```
bcftools view -s ^sample_to_exclude --trim-alt-alleles    my_file.vcf  
```

The error I get is:

```
bcftools: vcfutils.c:594: bcf_remove_allele_set: Assertion `nori==nA_ori' failed.
```

Both version 1.2 and 1.3 of bcftools give the same error.. 

I could send by email a small VCF file (less than a MB)  that can be used to reproduce and debug the issue. Which email could I send the file to? 

This issue / error might be related to these issues / errors:
https://github.com/samtools/bcftools/issues/322
https://github.com/samtools/bcftools/issues/256

Best, 

Neill

**Comment by pd3** [comment]
Send it to the email on my profile page.Thanks

**Comment by NeillGibson** [comment]
I have send you  an email with the small VCF file as attachment. Thank you for looking at the issue.

**Comment by pd3** [comment]
Thank you for the test case. The problem is that some fields have incorrect number of values. If you don't need them an easy fix is to drop them:

     bcftools annotate -x FMT/PL,FMT/AO,FMT/QA 

On our side, we'll fix this is by providing more informative error message in near future.

**Comment by NeillGibson** [comment]
Thank you for the diagnosis.
The causal issue for the invalid number of values is the use of vcfallelicprmitives with the --keep-geno option. 

Would bcftools accept a VCF file where the FMT/PL,FMT/AO,FMT/QA attributes are dropped only for the decomposed variants?  e.g. for the exclude sample trim alt alleles command?

Then I'll open a ticket at bcbio to include a step in the freebayes variant calling pipe to drop these attributes only for the the decomposed variants.

**Comment by mcshane** [comment]
This should be closed via samtools/htslib@ee4725892ee3f29a813ba906d7c8b07928a3f352. A more informative error message will be given and you can either fix the problem annotations or remove them with something like `bcftools annotate -x FMT/PL,FMT/AO,FMT/QA `
